### DISCUSSION TITLE
How do I add new auto-type sequences globally, for all programs/entries?

### DISCUSSION THREAD

**Discussion by LoganDark** [keepassxreboot/keepassxc#7983] [Q&A]
Websites commonly ask for you to repeat a new password twice in order to confirm it. Since KeePassXC always enters the password in perfectly (except for Biological Error), and I'm working through hundreds of accounts as part of a Google->KeePassXC migration, I would like to add a `{PASSWORD}{TAB}{PASSWORD}{ENTER}` sequence.

The problem is that KeePassXC only seems to support:

- defining one sequence to always be used in a particular window; or
- defining a sequence to be used for one specific entry

I want to instead add to the global pool of already-existing sequences like `{USERNAME}{TAB}{PASSWORD}{ENTER}`, `{USERNAME}`, `{USERNAME}{ENTER}`, `{PASSWORD}`, `{PASSWORD}{ENTER}`, `{TOTP}` and so on, so that it can be used for _all_ entries and _all_ applications, rather than just only one entry or only one window.

Is this possible to do?

**Comment by droidmonkey** [comment]
This is not possible. But with the new Auto-Type selection dialog in 2.7.x you can easily repeat an Auto-Type request and it will choose the last entry used (with 15 sec) and you can press Ctrl+2 to type the password.

**Reply by LoganDark** [reply]
> This is not possible.

Could you consider adding it, maybe as a group-level option? That way, you can either put it on root, or more specific groups.

> But with the new Auto-Type selection dialog in 2.7.x you can easily repeat an Auto-Type request and it will choose the last entry used (with 15 sec) and you can press Ctrl+2 to type the password.

Using keyboard shortcuts in KeePassXC causes the `{USERNAME}{TAB}{PASSWORD}{ENTER}` sequence to randomly stop working for some reason (it will do nothing when you select it), and I use that sequence a lot, so I'd rather not tempt fate lol (it's working right now)

**Reply by droidmonkey** [reply]
I'm referring to the global Auto-Type using the global keyboard shortcut. Not the entry level shortcut (Ctrl+Shift+V).

This type of feature would be outside the kdbx specification and kind of annoying to most users. They can obviously choose not to use it, but I certainly don't want a bunch of options added to all my entries. You can already define a different default typing pattern at the group level.

**Reply by LoganDark** [reply]
> I'm referring to the global Auto-Type using the global keyboard shortcut. Not the entry level shortcut (Ctrl+Shift+V)

I know what you're referring to. :/

> This type of feature would be outside the kdbx specification and kind of annoying to most users.

....What? But... KeePassXC already uses entry data for TOTP stuff. How would it annoy users when it doesn't even do anything unless you configure it differently?? You literally would have to edit a group and add new autotype sequences for anything to change :v

You know what, I'm not even gonna continue. You don't seem interested in this functionality at all (in fact, entirely opposed to it). I'd rather get my mind off it. FYI just because there's a workaround (don't add a custom sequence, just press the keys yourself!) doesn't mean it can't be made easier. I thought this was the purpose of auto type but apparently not.

**Reply by droidmonkey** [reply]
It's an interesting idea but will take a good effort to implement since it will be custom to keepassxc. 

There are other requests tracking a password change workflow which would be a better implementation for the specific use case mentioned at the beginning.
